[ERROR] vertex "vcf_instance.vcf_sddc" error: Invalid input

### Code of Conduct

- [x] I have read and agree to the [Code of Conduct](https://github.com/vmware/terraform-provider-vcf/blob/main/CODE_OF_CONDUCT.md).
- [ ] Vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue initial description to help the maintainers prioritize.
- [ ] Do not leave "+1" or other comments that do not add relevant information or questions.
- [ ] If you are interested in working on this issue or have submitted a pull request, please leave a comment.

### Terraform

v1.11.4

### Terraform Provider

v0.15.0

### VMware Cloud Foundation

5.2.0 and 5.2.1

### Description

Hi,
I am trying to bringup a new VCF stack using the terraform provider and the resource vcf_instance.

To get all the values, I just filled the deployment Excel file, converted it on the cloud builder into a json and transfered the values to terraform. 

When running terraform validate and terraform plan, I don't get any errors.

When running terraform apply, I directly get the message that there is a invalid input but the error is only referring to resource vcf_instance and not to a specific option.

Based on the error message I get on the cloud builder in the vcf-bringup.log, I tried multiple configs without success. 

I tried to remove all network sections and only left the management network, no success.
In my first draft, I tried to build this terraform script completely with variables but changed it to hardcoded values to prevent I made any mistakes with the call of the variables or value types.

So, I am a bit lost at the moment.

Regards
Benjamin


### Affected Resources or Data Sources

`resource vcf_instance`

### Terraform Configuration

```hcl
terraform {
  required_providers {
    vcf = {
      source  = "vmware/vcf"
      version = "0.15.0"
    }
  }
}

provider "vcf" {
  cloud_builder_host     = var.cloud_builder_hostname
  cloud_builder_username = var.cloud_builder_username
  cloud_builder_password = var.cloud_builder_password
}

resource "vcf_instance" "vcf_sddc" {
  instance_id                    = "v4-01"
  dv_switch_version              = "8.0.0"
  skip_esx_thumbprint_validation = true
  management_pool_name           = "v4-01-az1"
  ceip_enabled                   = false
  task_name = "Create SDDC"    

  sddc_manager {
    ip_address = "10.22.172.14"
    hostname   = "vcf01"

    root_user_credentials {
      username = "root"
      password = "*******************"
    }
    second_user_credentials {
      username = "vcf"
      password = "*******************"
    }
    local_user_password = "*******************"
  }

  ntp_servers = ["ntp1.noris.de", "ntp2.noris.de"]

  dns {
    domain                = "v4-01.nec.noris.de"
    name_server           = "62.128.1.42"
    secondary_name_server = "62.128.1.53"
  }

  network {
    subnet         = "10.22.173.0/26"
    vlan_id        = "3031"
    mtu            = "1500"
    network_type   = "MANAGEMENT"
    gateway        = "10.22.173.1"
    port_group_key = "cl01-nbg6c-v4-01-s1-mgmt"
  }

  network {
    subnet         = "10.22.172.0/24"
    vlan_id        = "3030"
    mtu            = "1500"
    network_type   = "VM_MANAGEMENT"
    gateway        = "10.22.172.1"
    port_group_key = "cl01-nbg-v4-01-s1-mgmt-vm"
  }

  network {
    subnet         = "10.22.174.64/26"
    include_ip_address_ranges {
      start_ip_address = "10.22.174.69"
      end_ip_address   = "10.22.174.125"
    }
    vlan_id        = "3034"
    mtu            = "9000"
    network_type   = "VSAN"
    gateway        = "10.22.174.65"
    port_group_key = "cl01-nbg6c-v4-01-s1-vsan"
  }

  network {
    subnet         = "10.22.174.0/26"
    include_ip_address_ranges {
      start_ip_address = "10.22.174.5"
      end_ip_address   = "10.22.174.61"
    }
    vlan_id        = "3033"
    mtu            = "9000"
    network_type   = "VMOTION"
    gateway        = "10.22.174.1"
    port_group_key = "cl01-nbg6c-v4-01-s1-vmotion-az1"
  }

  nsx {
    nsx_manager_size = "medium"

    nsx_manager {
      hostname = "nsxa"
      ip       = "10.22.172.11"
    }
    nsx_manager {
      hostname = "nsxb"
      ip       = "10.22.172.12"
    }
    nsx_manager {
      hostname = "nsxc"
      ip       = "10.22.172.13"
    }

    vip_fqdn                  = "nsx"
    vip                       = "10.22.172.10"
    root_nsx_manager_password = "*******************"
    nsx_admin_password        = "*******************"
    nsx_audit_password        = "*******************"

    overlay_transport_zone {
      zone_name    = "overlay-tz-nsx01"
      network_name = "overlay-net-nsx01"
    }

    transport_vlan_id = 3035

    ip_address_pool {
      name = "az1-nbg6c-v4-01-tep-pool"
      subnet {
        cidr    = "10.22.174.128/25"
        gateway = "10.22.174.129"
        ip_address_pool_range {
          start = "10.22.174.133"
          end   = "10.22.174.253"
        }
      }
    }
  }

  vsan {
    datastore_name = "van01-cl01-nbg-v4-01"
    esa_enabled    = true
  }

  dvs {
    mtu      = 9000
    dvs_name = "van01-cl01-nbg-v4-01-s1"
    vmnics   = ["vmnic0", "vmnic5"]
    networks = [
      "MANAGEMENT",
      "VSAN",
      "VMOTION",
      "VM_MANAGEMENT" 
    ]
  }

  dvs {
    mtu             = 9000
    is_used_by_nsxt = true
    dvs_name        = "van01-cl01-nbg-v4-01-s2"
    vmnics          = ["vmnic1", "vmnic2"]
    networks        = [ ]
    
  }

  cluster {
    cluster_name              = "van01-cl01-nbg-v4-01"
    cluster_evc_mode          = "intel-sapphirerapids"
    cluster_image_enabled     = true
    host_failures_to_tolerate = 1
  }

  psc {
    admin_user_sso_password = "*******************"
  }

  vcenter {
    vcenter_ip            = "10.22.172.9"
    vcenter_hostname      = "vc01"
    root_vcenter_password = "*******************"
    vm_size               = "medium"
  }

  host {
    credentials {
      username = "root"
      password = "*******************"
    }
    ip_address_private {
      ip_address = "10.22.173.4"
      gateway    = "10.22.173.1"
    }
    hostname    = "esx01-cl01-nbg6c"
    vswitch     = "vSwitch0"
    association = "v4-01.nec.noris.de"
  }

  host {
    credentials {
      username = "root"
      password = "*******************"
    }
    ip_address_private {
      ip_address = "10.22.173.5"
      gateway    = "10.22.173.1"
    }
    hostname    = "esx02-cl01-nbg6c"
    vswitch     = "vSwitch0"
    association = "v4-01.nec.noris.de"
  }

  host {
    credentials {
      username = "root"
      password = "*******************"
    }
    ip_address_private {
      ip_address = "10.22.173.6"
      gateway    = "10.22.173.1"
    }
    hostname    = "esx03-cl01-nbg6c"
    vswitch     = "vSwitch0"
    association = "v4-01.nec.noris.de"
  }

  host {
    credentials {
      username = "root"
      password = "*******************"
    }
    ip_address_private {
      ip_address = "10.22.173.7"
      gateway    = "10.22.173.1"
    }
    hostname    = "esx04-cl01-nbg6c"
    vswitch     = "vSwitch0"
    association = "v4-01.nec.noris.de"
  }
}
```

### Debug Output

```shell
Plan: 1 to add, 0 to change, 0 to destroy.

Do you want to perform these actions?
  Terraform will perform the actions described above.
  Only 'yes' will be accepted to approve.

  Enter a value: yes

2025-04-29T14:27:48.619Z [INFO]  backend/local: apply calling Apply
2025-04-29T14:27:48.623Z [INFO]  provider: configuring client automatic mTLS
2025-04-29T14:27:48.627Z [DEBUG] provider: starting plugin: path=.terraform/providers/registry.terraform.io/vmware/vcf/0.15.0/linux_amd64/terraform-provider-vcf_v0.15.0 args=[".terraform/providers/registry.terraform.io/vmware/vcf/0.15.0/linux_amd64/terraform-provider-vcf_v0.15.0"]
2025-04-29T14:27:48.627Z [DEBUG] provider: plugin started: path=.terraform/providers/registry.terraform.io/vmware/vcf/0.15.0/linux_amd64/terraform-provider-vcf_v0.15.0 pid=42274
2025-04-29T14:27:48.627Z [DEBUG] provider: waiting for RPC address: plugin=.terraform/providers/registry.terraform.io/vmware/vcf/0.15.0/linux_amd64/terraform-provider-vcf_v0.15.0
2025-04-29T14:27:48.631Z [INFO]  provider.terraform-provider-vcf_v0.15.0: configuring server automatic mTLS: timestamp=2025-04-29T14:27:48.631Z
2025-04-29T14:27:48.641Z [DEBUG] provider: using plugin: version=6
2025-04-29T14:27:48.641Z [DEBUG] provider.terraform-provider-vcf_v0.15.0: plugin address: address=/tmp/plugin2355120417 network=unix timestamp=2025-04-29T14:27:48.641Z
2025-04-29T14:27:48.667Z [WARN]  Provider "registry.terraform.io/vmware/vcf" produced an invalid plan for vcf_instance.vcf_sddc, but we are tolerating it because it is using the legacy plugin SDK.
    The following problems may be the cause of any confusing errors from downstream operations:
      - .network[0].teaming_policy: planned value cty.StringVal("loadbalance_loadbased") for a non-computed attribute
      - .network[1].teaming_policy: planned value cty.StringVal("loadbalance_loadbased") for a non-computed attribute
      - .network[2].teaming_policy: planned value cty.StringVal("loadbalance_loadbased") for a non-computed attribute
      - .network[3].teaming_policy: planned value cty.StringVal("loadbalance_loadbased") for a non-computed attribute
vcf_instance.vcf_sddc: Creating...
2025-04-29T14:27:48.668Z [INFO]  Starting apply for vcf_instance.vcf_sddc
2025-04-29T14:27:48.801Z [ERROR] provider.terraform-provider-vcf_v0.15.0: Response contains error diagnostic: diagnostic_severity=ERROR diagnostic_summary="Invalid input" tf_proto_version=6.8 tf_provider_addr=registry.terraform.io/vmware/vcf @caller=github.com/hashicorp/terraform-plugin-go@v0.26.0/tfprotov6/internal/diag/diagnostics.go:58 @module=sdk.proto tf_resource_type=vcf_instance tf_rpc=ApplyResourceChange diagnostic_detail="" tf_req_id=264c3b0c-6d94-2bd3-3e7d-66ba2a979dfc timestamp=2025-04-29T14:27:48.801Z
2025-04-29T14:27:48.810Z [ERROR] vertex "vcf_instance.vcf_sddc" error: Invalid input
╷
│ Error: Invalid input
│ 
│   with vcf_instance.vcf_sddc,
│   on vcf_instance.tf line 16, in resource "vcf_instance" "vcf_sddc":
│   16: resource "vcf_instance" "vcf_sddc" {
│ 
╵
2025-04-29T14:27:48.819Z [DEBUG] provider.stdio: received EOF, stopping recv loop: err="rpc error: code = Unavailable desc = error reading from server: EOF"
2025-04-29T14:27:48.820Z [INFO]  provider: plugin process exited: plugin=.terraform/providers/registry.terraform.io/vmware/vcf/0.15.0/linux_amd64/terraform-provider-vcf_v0.15.0 id=42274
2025-04-29T14:27:48.820Z [DEBUG] provider: plugin exited
```

On the Cloud Builder I can see in the `vcf-bringup.log`.

```shell
2025-04-29T14:18:16.154+0000 [bringup,6810dfa836ed06665fff384bccb95661,fd6c] ERROR [c.v.e.s.c.v.i.LocalizableAnnotationValidationUtil,http-nio-0.0.0.0-9080-exec-6] Spec violation SDDC_IP_ALLOCATION_IP_SUBNET_PATTERN_INVALID
2025-04-29T14:18:16.155+0000 [bringup,6810dfa836ed06665fff384bccb95661,fd6c] ERROR [c.v.e.s.c.v.i.LocalizableAnnotationValidationUtil,http-nio-0.0.0.0-9080-exec-6] Spec violation SDDC_IP_ALLOCATION_IP_SUBNET_PATTERN_INVALID
2025-04-29T14:18:16.155+0000 [bringup,6810dfa836ed06665fff384bccb95661,fd6c] ERROR [c.v.e.s.c.v.i.LocalizableAnnotationValidationUtil,http-nio-0.0.0.0-9080-exec-6] Spec violation SDDC_IP_ALLOCATION_IP_SUBNET_PATTERN_INVALID
2025-04-29T14:18:16.155+0000 [bringup,6810dfa836ed06665fff384bccb95661,fd6c] ERROR [c.v.e.s.c.v.i.LocalizableAnnotationValidationUtil,http-nio-0.0.0.0-9080-exec-6] Spec violation SDDC_IP_ALLOCATION_IP_SUBNET_PATTERN_INVALID
2025-04-29T14:18:16.155+0000 [bringup,6810dfa836ed06665fff384bccb95661,fd6c] ERROR [c.v.e.s.e.h.LocalizableRuntimeExceptionHandler,http-nio-0.0.0.0-9080-exec-6] [NQRFKN] REST_INVALID_API_INPUT Invalid input
com.vmware.evo.sddc.common.core.error.CompositeInvalidInputException: Invalid input
        at com.vmware.evo.sddc.exception.handler.MethodArgumentNotValidExceptionHandler.handleMethodArgumentNotValidException(MethodArgumentNotValidExceptionHandler.java:54)
        at jdk.internal.reflect.GeneratedMethodAccessor53.invoke(Unknown Source)
        at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at java.base/java.lang.reflect.Method.invoke(Method.java:568)
        at org.springframework.web.method.support.InvocableHandlerMethod.doInvoke(InvocableHandlerMethod.java:205)
        at org.springframework.web.method.support.InvocableHandlerMethod.invokeForRequest(InvocableHandlerMethod.java:150)
        at org.springframework.web.servlet.mvc.method.annotation.ServletInvocableHandlerMethod.invokeAndHandle(ServletInvocableHandlerMethod.java:118)
        at org.springframework.web.servlet.mvc.method.annotation.ExceptionHandlerExceptionResolver.doResolveHandlerMethodException(ExceptionHandlerExceptionResolver.java:413)
        at org.springframework.web.servlet.handler.AbstractHandlerMethodExceptionResolver.doResolveException(AbstractHandlerMethodExceptionResolver.java:74)
        at org.springframework.web.servlet.handler.AbstractHandlerExceptionResolver.resolveException(AbstractHandlerExceptionResolver.java:141)
        at org.springframework.web.servlet.handler.HandlerExceptionResolverComposite.resolveException(HandlerExceptionResolverComposite.java:80)
        at org.springframework.web.servlet.DispatcherServlet.processHandlerException(DispatcherServlet.java:1341)
        at org.springframework.web.servlet.DispatcherServlet.processDispatchResult(DispatcherServlet.java:1152)
        at org.springframework.web.servlet.DispatcherServlet.doDispatch(DispatcherServlet.java:1098)
        at org.springframework.web.servlet.DispatcherServlet.doService(DispatcherServlet.java:974)
        at org.springframework.web.servlet.FrameworkServlet.processRequest(FrameworkServlet.java:1014)
        at org.springframework.web.servlet.FrameworkServlet.doPost(FrameworkServlet.java:914)
        at jakarta.servlet.http.HttpServlet.service(HttpServlet.java:590)
        at org.springframework.web.servlet.FrameworkServlet.service(FrameworkServlet.java:885)
        at jakarta.servlet.http.HttpServlet.service(HttpServlet.java:658)
        at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:205)
        at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:149)
        at org.apache.tomcat.websocket.server.WsFilter.doFilter(WsFilter.java:51)
        at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:174)
        at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:149)
        at org.springframework.web.filter.RequestContextFilter.doFilterInternal(RequestContextFilter.java:100)
        at org.springframework.web.filter.OncePerRequestFilter.doFilter(OncePerRequestFilter.java:116)
        at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:174)
        at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:149)
        at org.springframework.web.filter.FormContentFilter.doFilterInternal(FormContentFilter.java:93)
        at org.springframework.web.filter.OncePerRequestFilter.doFilter(OncePerRequestFilter.java:116)
        at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:174)
        at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:149)
        at org.springframework.web.filter.ServerHttpObservationFilter.doFilterInternal(ServerHttpObservationFilter.java:109)
        at org.springframework.web.filter.OncePerRequestFilter.doFilter(OncePerRequestFilter.java:116)
        at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:174)
        at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:149)
        at org.springframework.web.filter.CharacterEncodingFilter.doFilterInternal(CharacterEncodingFilter.java:201)
        at org.springframework.web.filter.OncePerRequestFilter.doFilter(OncePerRequestFilter.java:116)
        at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:174)
        at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:149)
        at org.apache.catalina.core.StandardWrapperValve.invoke(StandardWrapperValve.java:167)
        at org.apache.catalina.core.StandardContextValve.invoke(StandardContextValve.java:90)
        at org.apache.catalina.authenticator.AuthenticatorBase.invoke(AuthenticatorBase.java:482)
        at org.apache.catalina.core.StandardHostValve.invoke(StandardHostValve.java:115)
        at org.apache.catalina.valves.ErrorReportValve.invoke(ErrorReportValve.java:93)
        at org.apache.catalina.core.StandardEngineValve.invoke(StandardEngineValve.java:74)
        at org.apache.catalina.connector.CoyoteAdapter.service(CoyoteAdapter.java:344)
        at org.apache.coyote.http11.Http11Processor.service(Http11Processor.java:391)
        at org.apache.coyote.AbstractProcessorLight.process(AbstractProcessorLight.java:63)
        at org.apache.coyote.AbstractProtocol$ConnectionHandler.process(AbstractProtocol.java:896)
        at org.apache.tomcat.util.net.NioEndpoint$SocketProcessor.doRun(NioEndpoint.java:1744)
        at org.apache.tomcat.util.net.SocketProcessorBase.run(SocketProcessorBase.java:52)
        at org.apache.tomcat.util.threads.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1191)
        at org.apache.tomcat.util.threads.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:659)
        at org.apache.tomcat.util.threads.TaskThread$WrappingRunnable.run(TaskThread.java:63)
        at java.base/java.lang.Thread.run(Thread.java:840)
```

### Panic Output

_No response_

### Expected Behavior

Terraform should deploy a new SDDC manager with a managent Workload Domain 

### Actual Behavior

Stops the terraform apply workflow with the message "invalid input".

### Steps to Reproduce

I pasted my terraform script in "terraform config". This is the way I do it.

### Environment Details

_No response_

### Screenshots

_No response_

### References

_No response_

Hey, 

thanks for your fast response :)

I think I had in one of my versions as "netmask" befor I've realised there is also a subnet_mask option in this network block.

But I tried it again, just to be sure.

`  network {
    subnet        = "255.255.255.192"
    vlan_id        = "3030"
    mtu            = "1500"
    network_type   = "VM_MANAGEMENT"
    gateway        = "10.22.172.1"
    port_group_key = "cl01-nbg-v4-01-s1-mgmt-vm"
  }`

I get still the same error

`2025-04-30T05:48:52.766Z [ERROR] provider.terraform-provider-vcf_v0.15.0: Response contains error diagnostic: diagnostic_severity=ERROR diagnostic_summary="Invalid input" tf_provider_addr=registry.terraform.io/vmware/vcf tf_resource_type=vcf_instance tf_proto_version=6.8 tf_req_id=9ce993d4-db58-f1ee-119e-1a61d78ec8a6 tf_rpc=ApplyResourceChange @caller=github.com/hashicorp/terraform-plugin-go@v0.26.0/tfprotov6/internal/diag/diagnostics.go:58 @module=sdk.proto diagnostic_detail="" timestamp=2025-04-30T05:48:52.766Z
2025-04-30T05:48:52.770Z [ERROR] vertex "vcf_instance.vcf_sddc" error: Invalid input
╷
│ Error: Invalid input
│ 
│   with vcf_instance.vcf_sddc,
│   on vcf_instance.tf line 16, in resource "vcf_instance" "vcf_sddc":
│   16: resource "vcf_instance" "vcf_sddc" {
│ 
╵`

On cloud builder 
`2025-04-30T05:48:52.755+0000 [bringup,6811b9c4fa1b99cc6dd9cbaaa6b8df47,df8c] ERROR [c.v.e.s.c.v.i.LocalizableAnnotationValidationUtil,http-nio-0.0.0.0-9080-exec-10] Spec violation SDDC_IP_ALLOCATION_IP_SUBNET_PATTERN_INVALID
2025-04-30T05:48:52.756+0000 [bringup,6811b9c4fa1b99cc6dd9cbaaa6b8df47,df8c] ERROR [c.v.e.s.c.v.i.LocalizableAnnotationValidationUtil,http-nio-0.0.0.0-9080-exec-10] Spec violation SDDC_IP_ALLOCATION_IP_SUBNET_PATTERN_INVALID
2025-04-30T05:48:52.756+0000 [bringup,6811b9c4fa1b99cc6dd9cbaaa6b8df47,df8c] ERROR [c.v.e.s.c.v.i.LocalizableAnnotationValidationUtil,http-nio-0.0.0.0-9080-exec-10] Spec violation SDDC_IP_ALLOCATION_IP_SUBNET_PATTERN_INVALID
2025-04-30T05:48:52.756+0000 [bringup,6811b9c4fa1b99cc6dd9cbaaa6b8df47,df8c] ERROR [c.v.e.s.c.v.i.LocalizableAnnotationValidationUtil,http-nio-0.0.0.0-9080-exec-10] Spec violation SDDC_IP_ALLOCATION_IP_SUBNET_PATTERN_INVALID
2025-04-30T05:48:52.756+0000 [bringup,6811b9c4fa1b99cc6dd9cbaaa6b8df47,df8c] ERROR [c.v.e.s.e.h.LocalizableRuntimeExceptionHandler,http-nio-0.0.0.0-9080-exec-10] [JLPETV] REST_INVALID_API_INPUT Invalid input
com.vmware.evo.sddc.common.core.error.CompositeInvalidInputException: Invalid input
`

I also tried this version
`  network {
    subnet_mask    = "255.255.255.192"
    vlan_id        = "3030"
    mtu            = "1500"
    network_type   = "VM_MANAGEMENT"
    gateway        = "10.22.172.1"
    port_group_key = "cl01-nbg-v4-01-s1-mgmt-vm"
  }`

--- a/internal/sddc/sddc_host_subresource.go
+++ b/internal/sddc/sddc_host_subresource.go
@@ -123,13 +123,17 @@ func getIPAllocationBindingFromSchema(rawData []interface{}) *vcf.IpAllocation {
 	cidr := data["cidr"].(string)
 	gateway := data["gateway"].(string)
 	ipAddress := data["ip_address"].(string)
-	subnet := data["subnet"].(string)
+
+	var subnet *string
+	if data["subnet"].(string) != "" {
+		subnet = utils.ToStringPointer(data["subnet"])
+	}
 
 	ipAllocationBinding := &vcf.IpAllocation{
 		Cidr:      &cidr,
 		Gateway:   &gateway,
 		IpAddress: ipAddress,
-		Subnet:    &subnet,
+		Subnet:    subnet,
 	}
 	return ipAllocationBinding
 }

